### PR TITLE
Refactor scaling code

### PR DESCRIPTION
**Pull request by last-partizan**
Basic idea here is to remove scaling from as much app as possible.

Let's make only fonts aware of scaling, everything else is expected to just works.

What do you think? Is it good idea? Could this give us any problems in the future?

The basics are already working, it scales as i expect it to. No artifacts.

## What needs to be adressed before merge:

- [x] Fira Code font problem.
    It is not mac-specific, to reproduce you need to have `Fira Code` installed, and no `guifont` in vimrc. Then neovide tries to load Fira Code as default font, and with scaling greater than 1, bug happens.

-------------------

## What kind of change does this PR introduce?
- Fix
- Refactor

## Did this PR introduce a breaking change?
_A breaking change includes anything that breaks backwards compatibility either at compile or run time._
- No

### CODE REVIEW

File: src/window/window_wrapper/mod.rs [hunk 1]
```diff
@@ -57,22 +56,7 @@ impl GlutinWindowWrapper {
         let window = self.windowed_context.window();
         if self.fullscreen {
             window.set_fullscreen(None);
-
-            // Use cached size and position
-            window.set_inner_size(LogicalSize::new(
-                self.cached_size.width,
-                self.cached_size.height,
-            ));
-            window.set_outer_position(LogicalPosition::new(
```
Comment (Kethku) [inline src/window/window_wrapper/mod.rs:66]: Looks like this got removed. Was this intentional? I don't remember off the top of my head what the state is for cached window size

File: src/window/window_wrapper/mod.rs [hunk 2]
```diff
@@ -166,52 +150,49 @@ impl GlutinWindowWrapper {
 
     pub fn draw_frame(&mut self, dt: f32) {
         let window = self.windowed_context.window();
-        let scaling = window.scale_factor();
-        let new_size = window.inner_size();
+        let scale_factor = 1.0; // window.scale_factor();
```
Comment (Kethku) [inline src/window/window_wrapper/mod.rs:153]: Is the scale factor here supposed to be commented out?

File: src/window/window_wrapper/mod.rs (hunk 1 above)
Comment (last-partizan) [reply src/window/window_wrapper/mod.rs:66]: Oh, that's fullscreen fix from earlier PR, probably i forgot to rebase.

File: src/window/window_wrapper/mod.rs (hunk 2 above)
Comment (last-partizan) [reply src/window/window_wrapper/mod.rs:153]: That's temporary. I'll remove it alltogether and use PhysicalSize.

**Comment by last-partizan** [comment]
I think it's ready for review and testing.

Bet let's not rush with merging. This is big refactoring and it needs to be reviewed carefully.

**Comment by vivlim** [comment]
The fonts are messed up in retina scaling mode on my mac:
image

I get this message with any retina resolution set
```
INFO [neovide::window::window_wrapper] window created (scale_factor: 2, font_size: 17x35)
```

If I switch to 1x scaling then the font is fine and I get this
```
INFO [neovide::window::window_wrapper] window created (scale_factor: 1, font_size: 8x18)
```

**Comment by last-partizan** [comment]
@vivlim what fonts are you using?

**Comment by vivlim** [comment]
Fira Code, but this also reproduces if I bypass my nvim config entirely.

**Comment by last-partizan** [comment]
@vivlim i was able to reproduce it on mac, with empty config. But as soon as i set guifont to something like

```
set guifont=JetBrainsMono\ Nerd\ Font:h15
```
or
```
set guifont=monospace
```

Issue is gone.

Could you show your exact guifont line with `Fira Code`, and how do you installed it?

**Comment by last-partizan** [comment]
@vivlim please check again, it must work now as expected.

**Comment by vivlim** [comment]
Yep, that is working great now.

**Comment by Kethku** [comment]
I just looked, things seem a little weird on windows. Almost as though the scaling is larger than expected... Capturing my findings here so that I can dig in more outside of work hours

At 100% scale, the font looks perfect. And in fact matches the expected size
image

However at 150% scale, the neovide text looks too small by maybe 30% or so 
image

At 200% it seems worse
image

And 250% even further
image

I think this means that on windows the scaling isn't taken into account as the size of the text doesn't seem to change. I'm wondering if winit isn't reporting a change in scale.

A related issue is that the window changes size drastically by the scaling factor eg the window increases in size 250% at scaling 250% when moved from a 100% scaling window.

**Comment by last-partizan** [comment]
If you're changing scale and not restarting neovide, this is expected. Right now scaling is requested only once at the app start.

**Comment by Kethku** [comment]
Oh gotcha. I think this is probably important to change for setups with multiple different scale factors. But I can make that change after this merges. If thats the case, I think this works! Great stuff

**Comment by last-partizan** [comment]
I have few other refactorings in mind, and can look into dymanic scaling as well after merge.

Also, i have no other ideas what else to fix or check in this branch, so let's merge it.

**Comment by last-partizan** [comment]
Please, use `Squash and merge`, and commit message in conventional commits format.

**Pull request merged**